lib/Sema/CSDiagnostics.cpp

```diff
@@ -1222,17 +1228,18 @@ ASTNode InvalidCoercionFailure::getAnchor() const {
 bool InvalidCoercionFailure::diagnoseAsError() {
   auto fromType = getFromType();
   auto toType = getToType();
+  auto *CE = getAsExpr<CoerceExpr>(getRawAnchor());
 
   emitDiagnostic(diag::cannot_coerce_to_type, fromType, toType);
 
   if (UseConditionalCast) {
     emitDiagnostic(diag::missing_optional_downcast)
-        .highlight(getSourceRange())
-        .fixItReplace(getLoc(), "as?");
```

Comment:
Sorry, didn't notice this initially... Should `getLoc()` of `InvalidCoercionFailure` be re-defined instead?

Reply:
Fixed!